### PR TITLE
Translate authentication.md to german

### PR DESCRIPTION
**Pull request by mezotv**
Adds the correct and finished German translation for the authentication.md file. Also fixes the index.de.md translation!

This is the complete and improved version of #827

### CODE REVIEW

**Comment by mezotv** [comment]
If I missed anything let me know and I will add that as soon as possible!

**Comment by mezotv** [comment]
Hey @ptoffy would you be able to check this out?

**Review by ptoffy — DISMISSED** [review]
I reviewed until line ~100 and have found quite some missing pieces. Please review the file and fix the pieces you can find, then I'm happy to fix the left overs! (If @mattesmohr isn't faster 😄)

**Comment by mezotv** [comment]
> I reviewed until line ~100 and have found quite some missing pieces. Please review the file and fix the pieces you can find, then I'm happy to fix the left overs! (If @mattesmohr isn't faster 😄)

Yea I didn't check the first half seems like matt left a bit out ill get that fixed!

File: docs/security/authentication.de.md [hunk 1]
```diff
@@ -0,0 +1,874 @@
...
+
+## Einführung
+
+Vapor ermöglicht die Basis- und Bearerauthentifzierung mittels dem Authorization-Header. Die Authenifzierung erfolgt durch einen sogenannten _Authenticator_, der die eigentliche Logik beinhaltet und dazu verwendent wird um, einzelne Endpunkte oder auch die gesamte Anwendung zu sichern.
+| Protokoll                                                   | Beschreibung                                             |
+|-------------------------------------------------------------|----------------------------------------------------------|
+| `RequestAuthenticator`/`AsyncRequestAuthenticator`          | Base authenticator capable of creating middleware.       |
+| [`BasicAuthenticator`/`AsyncBasicAuthenticator`](#basis)    |Authenticates Basic authorization header.                 |
+| [`BearerAuthenticator`/`AsyncBearerAuthenticator`](#bearer) |Authenticates Bearer authorization header.                |
+| `CredentialsAuthenticator`/`AsyncCredentialsAuthenticator`  |Authenticates a credentials payload from the request body.|
```
[the remark this answers is not in the document]
Comment (mezotv) [reply docs/security/authentication.de.md]: Resolved!

**Comment by mezotv** [comment]
@ptoffy I have added the necessary changes let me know if you find something!

**Comment by mattesmohr** [comment]
Hiya Dominik, I hope you don't mind I hope in. Paul and I have talked and we think I should do the review, since I have started the initial translation for the authentication.md. First, thank you for taking the time to finish the translation. I appreciate it. I will take a look at least at the end of the week, as I write you in between two appointments right now and the rest of the week doesn't look different.

Since this is about the German translation, I think it is totally fine to express ourself in German.

**Comment by ptoffy** [comment]
Hey @mattesmohr, do you have any update on this? :)

**Comment by mattesmohr** [comment]
> Hey @mattesmohr, do you have any updates on this? :)

Yes and no. I am back from vacation and I have some time to spare. Will finish the review this week.

File: docs/security/authentication.de.md [hunk 2]
```diff
@@ -0,0 +1,880 @@
...
+
+```swift
+let protectedRoutes = app.grouped([User.SessionAuthenticator(), redirecteMiddleware])
+```
+
+### Formularanfrage
+
+Für die Authentifizierung und spätere Sitzungen, die Anmeldung erfolgt meistens über ein Formular.
+
+Um einen Benutzer und zukünftige Anfragen mit einer Sitzung zu authentifizieren, musst du einen Benutzer anmelden. Vapor bietet ein `ModelCredentialsAuthenticatable`-Protokoll, dem du entsprechen musst. Damit kannst du dich über ein Formular anmelden. Zuerst musst du deinen "User" an dieses Protokoll anpassen:
```
Comment (mattesmohr) [inline docs/security/authentication.de.md:765]:
```suggestion
Um einen Benutzer und bestehende Sitzungen zu authentifizieren, muss sich zuerst ein Benutzer anmelden. Vapor stellt uns für die Anmeldungsabwicklung das Protokoll _ModelCredentialsAuthenticatable_ zur Verfügung, mit das wir unser Objekt _User_ versehen. 
```

**Review by mattesmohr — CHANGES_REQUESTED** [review]
Anbei die erste Hälfte des Reviews. Mehr habe ich heute nicht geschafft.

**Comment by mezotv** [comment]
> Anbei die erste Hälfte des Reviews. Mehr habe ich heute nicht geschafft.

Sehr cool danke! Leider hab ich kaum zeit mehr werde mir das ganze also erst mitte November anschauen können.

File: docs/index.de.md [hunk 3]
```diff
@@ -25,4 +25,4 @@ Die Dokumentationen zu älteren Versionen von Vapor findest du unter [https://le
 
 ## Authoren
 
-Das Vapor-Kernteam und viele weitere Mitglieder der Vapor-Community.
+Das Vapor Kernteam und viele andere Mitglieder der Vapor Community.
```
[the remark this answers is not in the document]
Comment (mezotv) [reply docs/index.de.md:28]: Gerne, kann ich machen!

File: docs/security/authentication.de.md [hunk 4]
```diff
@@ -0,0 +1,880 @@
+# Authentifizierung
+
+Bei der Authentifizierung handelt es sich um die Überprüfung einer Benutzeridentität, zum Beispiel anhand von Anmeldeinformationen oder einem Token. Die Authentifizierung unterscheidet sich von der Autorisierung, bei der die Berechtigung eines zuvor authentifizierten Benutzers zur Durchführung bestimmter Aufgaben überprüft wird.
+
+
+## Einführung
+
+Vapor ermöglicht die [Basis](https://tools.ietf.org/html/rfc7617)- und [Bearer](https://tools.ietf.org/html/rfc6750) Authentifizierung mittels dem `Authorization` Header. Man kann auch einen User mit den Daten die in der [Content](../basics/content.de.md) API sind authentifizieren.
```
[the remark this answers is not in the document]
Comment (mezotv) [reply docs/security/authentication.de.md:8]: Arbeite leider nicht mit Vapor darum kann ich hier nicht viel helfen.

File: docs/security/authentication.de.md [hunk 5]
```diff
@@ -0,0 +1,878 @@
...
+}
+```
+
+Im folgenden Beispiel folgt das Objekt _UserTokenAuthenticator_ dem _UserPasswordAuthenticator_ und zum Schluss die _GuardMiddleware_.
+
+In diesem Beispiel wird von zwei Authentifikatoren `UserPasswordAuthenticator` und `UserTokenAuthenticator` ausgegangen, die beide `User` authentifizieren. Diese beiden Authentifikatoren werden der Routengruppe hinzugefügt. Schließlich wird `GuardMiddleware` nach den Authentifikatoren hinzugefügt, um sicherzustellen, dass `User` erfolgreich authentifiziert wurde.
+
+Die Zusammenstellung der Authentikator macht es möglich, dass der Benutzer sich 
+
+Diese Zusammensetzung von Authentifikatoren führt zu einer Route, auf die entweder mit einem Passwort oder einem Token zugegriffen werden kann. Ein solcher Weg könnte es einem Benutzer ermöglichen, sich anzumelden und ein Token zu generieren, das er dann weiter zur Generierung neuer Token verwenden kann.
```
Comment (mattesmohr) [inline docs/security/authentication.de.md:186]:
```suggestion
Die Kombination aus den beiden oben genannten Authentifikatoren ermöglicht den Zugriff mit einem Passwort oder einem Token. Auf dieser Weise kann sich ein Benutzer anmelden um wiederum regelmäßig einen Token zu generieren.
```

**Review by mattesmohr — CHANGES_REQUESTED** [review]

**Comment by mezotv** [comment]
Any update on this @mattesmohr @ptoffy ?ñ

**Comment by mattesmohr** [comment]
@mezotv  Hiya, ich bin nicht ganz zufrieden, werde aber noch mit meiner Korrektur auf meiner Festplatte etwas brauchen. Ein, zwei Abschnitte sind noch übrig. Damit das PR mal erledigt wird, werde ich es, wenn die Test durchgelaufen und bestanden sind, nachher oder heute Abend mergen.

**Review by mattesmohr — APPROVED** [review]

**Pull request merged**